subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/IncrementalCompilerDecorator.java

```diff
@@ -71,8 +81,13 @@ public IncrementalCompilerDecorator(JarClasspathSnapshotMaker jarClasspathSnapsh
             LOG.info("{} - is not incremental. Unable to infer the source directories.", displayName);
             return cleaningCompiler;
         }
-        if (!annotationProcessorPath.isEmpty()) {
-            LOG.info("{} - is not incremental. Annotation processors are present.", displayName);
+        List<AnnotationProcessorDeclaration> nonIncrementalProcessors = getNonIncrementalProcessors();
+        if (!nonIncrementalProcessors.isEmpty()) {
+            StringBuilder processorListing = new StringBuilder();
```

Comment:
:x: You are building a message string that is not going to be used if the log level is not INFO. I'd suggest to wrap this into a `if (logger.isInfoEnabled)` block.

Reply:
Fixed